hilti/toolchain/include/ast/operators/port.h

```diff
@@ -2,15 +2,25 @@
 
 #pragma once
 
+#include <vector>
+
 #include <hilti/ast/builder/type.h>
 #include <hilti/ast/operators/common.h>
+#include <hilti/ast/types/enum.h>
+#include <hilti/ast/types/integer.h>
 #include <hilti/ast/types/port.h>
 
 namespace hilti::operator_ {
 
 STANDARD_OPERATOR_2(port, Equal, type::Bool(), type::Port(), type::Port(), "Compares two port values.")
 STANDARD_OPERATOR_2(port, Unequal, type::Bool(), type::Port(), type::Port(), "Compares two port values.")
 
+BEGIN_KEYWORD_CTOR(port, Ctor, "port", type::Port(), "Creates a port instance.")
+    std::vector<Operand> parameters() const {
+        return {{"port", hilti::type::UnsignedInteger(16)}, {"protocol", type::Enum(type::Wildcard())}};
```

Comment:
I was wondering why we weren't more restrictive with the port type arg like e.g.,
```suggestion
         return {{"port", hilti::type::UnsignedInteger(16)}, {"protocol", builder::typeByID("hilti::Protocol")}};```
```

When I tried this the `spicy.types.port.ops` passed, but `hilti.types.port.ops` failed,

```console
$ btest -dv hilti.types.port.ops
hilti.types.port.ops ...
  > ${HILTIC} -j %INPUT >output
... hilti.types.port.ops failed
  % '${HILTIC} -j /Users/bbannier/src/spicy/tests/.tmp/hilti.types.port.ops/ops.hlt >output' failed unexpectedly (exit code 1)
  % cat .stderr
  [error] /Users/bbannier/src/spicy/tests/.tmp/hilti.types.port.ops/ops.hlt:14:14: unknown ID 'p3'
  [error] /Users/bbannier/src/spicy/tests/.tmp/hilti.types.port.ops/ops.hlt:17:14: unknown ID 'p3'
  [error] /Users/bbannier/src/spicy/tests/.tmp/hilti.types.port.ops/ops.hlt:21:8: unknown ID 'p3'
  [error] /Users/bbannier/src/spicy/tests/.tmp/hilti.types.port.ops/ops.hlt:25:8: unknown ID 'p3'
  [error] /Users/bbannier/src/spicy/tests/.tmp/hilti.types.port.ops/ops.hlt:10:13: unsupported operator: port(<uint<64>>, <hilti::Protocol>)
  [error] hiltic: aborting after errors

1 of 1 test failed

Reply:
Yeah, this is a shortcoming of our resolver: `typeByID` creates an `UnresolvedID`, which we'd never resolve because it's not part of any AST; and then the resolver doesn't know what to do with it for matching against it.